pkg/auth/multiauthenticator.go

```diff
@@ -5,53 +5,41 @@ import (
 	"fmt"
 )
 
-type IssuerWithClientID struct {
-	// Issuer of the token. The issuer cannot be empty.
-	Issuer string
-	// ClientID of the token. The clientID cannot be empty.
-	ClientID string
+type multiAuthenticatorImpl struct {
+	authenticators map[string][]Authenticator
 }
 
-type multiAuthenticatorWithClientIDImpl struct {
-	authenticators map[IssuerWithClientID]Authenticator
-}
-
-func NewMultiAuthenticatorWithClientID(
-	authenticators map[IssuerWithClientID]Authenticator) (MultiAuthenticatorWithClientID, error) {
-	authMap := make(map[IssuerWithClientID]Authenticator)
-	for issuerWithClientID, authenticator := range authenticators {
-		if issuerWithClientID.ClientID == "" {
-			return nil, fmt.Errorf("clientID cannot be empty")
-		}
-		if issuerWithClientID.Issuer == "" {
-			return nil, fmt.Errorf("issuer cannot be empty")
+// NewMultiAuthenticator maintains a list of authenticators for a given issuer.
+// The input argument is a map of issuers to a list of authenticators for that issuer.
+// NOTE: This interface does not check if there are duplicate authenticators.
+func NewMultiAuthenticator(
+	authenticators map[string][]Authenticator,
+) (MultiAuthenticatorWithClientID, error) {
+	authMap := make(map[string][]Authenticator)
+	for issuer, authenticatorsList := range authenticators {
+		if len(authenticatorsList) == 0 {
+			return nil, fmt.Errorf("empty authenticators list for issuer %v", issuer)
 		}
-		authMap[issuerWithClientID] = authenticator
+		authMap[issuer] = authenticatorsList
```

Comment:
nit: we should be making a copy of the slice as well. But since this is getting tricky, maybe we can just add a comment to the function saying that the input map is considered owned by framework and not make a copy of either the map or the slice.